### PR TITLE
aspect: add --video-unscaled=downscale-big

### PR DESCRIPTION
**Pull request by rr-**
Solves #3423.

### CODE REVIEW

File: video/out/aspect.c [hunk 1]
```diff
@@ -54,6 +54,13 @@ static void aspect_calc_panscan(struct mp_vo_opts *opts,
         fwidth = w * monitor_par;
         fheight = h;
         vo_panscan_area = 0;
+        if (unscaled == 2 && (fwidth > window_w || fheight > window_h)) {
+            aspect_calc_panscan(opts,
+                                w, h, d_w, d_h, 0,
+                                window_w, window_h, monitor_par,
+                                out_w, out_h);
+            return;
+        }
```
Comment (wm4) [inline video/out/aspect.c]: No, I don't like this.

Comment (rr-) [inline video/out/aspect.c]: Can you elaborate? Thanks

Comment (wm4) [inline video/out/aspect.c]: Calling it recursively again seems far too messy.

Comment (rr-) [inline video/out/aspect.c]: True, refactored and pushed

File: DOCS/man/options.rst [hunk 2]
```diff
@@ -738,10 +738,12 @@ Video
...
     but seems to be correct in other players.
 
-``--video-unscaled``
+``--video-unscaled=<no|yes|downscale-big>``
     Disable scaling of the video. If the window is larger than the video,
-    black bars are added. Otherwise, the video is cropped. The video still
-    can be influenced by the other ``--video-...`` options.
+    black bars are added. Otherwise, the video is cropped, unless the option
+    is set to ``downscale-big``, in which case the video is fit to window as
+    usual. The video still can be influenced by the other ``--video-...``
```
Comment (rr-) [inline DOCS/man/options.rst]: I haven't been able to figure what `vo_panscan_area` is for, I left the assignment as it was introduced in b968d779

Comment (rr-) [inline DOCS/man/options.rst]: Added the note in a separate commit, since that logic was already in place before this PR

**Comment by wm4** [comment]
Your change disables panscan, even though your manpage additions say: 
```
Otherwise, the video is cropped, unless the option is set to ``downscale-big``, in which case the video is fit to window as usual.
 ```

This is not true. You always disable panscan, even if the image is supposedly displayed normally.

**Comment by rr-** [comment]
I don't understand,

>Your change disables panscan

It was disabled before:

https://github.com/mpv-player/mpv/pull/3436/files#diff-206d8efa80bacfb748475687876890f6L56

>You always disable panscan, even if the image is supposedly displayed normally.

Hence this note:

https://github.com/mpv-player/mpv/pull/3436/commits/8d9490290ac163f899fe8742d28ef1a2ffab9731#diff-1d6a5494d6e6ca1cadd10688a1e233bdR748

**Comment by wm4** [comment]
While you're technically right that --video-unscaled with values other than "no" disable panscan, it's not consistent with the manpage description "the video is fit to window as usual".

**Comment by rr-** [comment]
Dropped `as usual` in the hope it's less confusing now, and sorry for prolonging this PR

**Comment by wm4** [comment]
I still question why you'd disable panscan in this case. That seems odd?

Also minor improvement: you could put `w *= monitor_par;` at the start of the function to remove some minor duplication.

**Comment by rr-** [comment]
But wouldn't it be weird if panscan sometimes worked and sometimes didn't during one session? Esp. that it influences the area that's used to decide whether to fit the frame or not.

Fixed duplication

**Comment by wm4** [comment]
Well, whatever. I'll merge it, but I might change the panscan thing later.

**Comment by kevinlekiller** [comment]
@rr- is it possible to add some kind of minimum to `--video-unscaled=downscale-big`

For example : if the display is 1920x1080, you play a 1900x1000 video, disable scaling, you then play a 1280x720 video, enable scaling.

The minimum could be a percentage or something, for example, disable scaling if at least 90% of 1920x1080, (between 1728x972 and 1920x1080).

Thanks.

**Comment by wm4** [comment]
Please no. We can't hardcode every small use-case into options. Use scripting.

**Comment by rr-** [comment]
I agree with @wm4, I only opened this PR and the original issue because any scaling done by the scripts bears inherent flickering when changing files, and this is painful mostly when navigating through image playlists. A 0.2-0.3s flicker on opening shouldn't hurt with normal videos.

@kevinlekiller I believe you could adapt my old image zoom script (the one that caused flickering) for videos: https://github.com/rr-/dotfiles/blob/b8ef36445fb4ab2598c1159906990fbf516e2dfc/mod-mpv/config/images/fancy_zoom.lua

**Comment by kevinlekiller** [comment]
It's no problem, thanks anyways!, I already had a script to do it, just wondered if I could remove another script if it was something simple to do (starting to have a lot of scripts).

**Pull request closed without merging**